addons/website_sale/controllers/main.py

```diff
@@ -160,16 +160,21 @@ def _get_search_order(self, post):
         # id is added to be sure that order is a unique sort key
         return 'is_published desc,%s , id desc' % post.get('order', 'website_sequence desc')
 
-    def _get_search_domain(self, search, category, attrib_values):
-        domain = request.website.sale_product_domain()
+    def _get_search_domain(self, search, category, attrib_values, search_in_description):
+        domains = [request.website.sale_product_domain()]
         if search:
             for srch in search.split(" "):
-                domain += [
-                    '|', '|', '|', ('name', 'ilike', srch), ('description', 'ilike', srch),
-                    ('description_sale', 'ilike', srch), ('product_variant_ids.default_code', 'ilike', srch)]
+                subdomains = [
+                    [('name', 'ilike', srch)],
+                    [('product_variant_ids.default_code', 'ilike', srch)]
+                ]
+                if search_in_description:
+                    subdomains.append([('description', 'ilike', srch)])
+                    subdomains.append([('description_sale', 'ilike', srch)])
+                domains.append(expression.OR(subdomains))
```

Comment:
Why change all the other lines on this method? Couldn't you just += here too and not make domain an array of array?

Reply:
will be no more readable, since you need to preppenr | beofre the 2 others searches

base_domain = ,'|', cond1, cond2
if search_Desc:
   base domain = '|','|', + base_domain + ' cond3, cond4 

Solutions proposed here to use the expression.OR lgtm

Reply:
Indeed OR is good, I mean use `domain += expression.OR(subdomains)`

Changing all the other += to append is unnecessary I believe.

Reply:
Unnecessary yes but way better to me :)